### ISSUE TITLE
LS-GAN Target and Multi-Level training

### ISSUE THREAD

**Issue by Nadavc220** [wasidennis/AdaptSegNet#88]
Hi,
Thanks for this contribution, the code is very easy to read and use.

My questions are:
1) The LS-GAN result of 44.1% mIoU was achieved by using a single-level module? if so what level was used, feature or output?
2) Is there any problem training a multi-level net with the LS-GAN loss target just by adding --gan LS as you did with the single-level case?
3)  I see that lambda-adv-target2 was changed too in the process of using the LS-GAN mode. Does this mean there is some hyper-parameter search to be done in order to use the LS-GAN with a multi-level model training process?

Thanks again. =)

**Comment by wasidennis** [comment]
Thanks @Nadavc220 for the nice feedback.

1. Yes, we only use the single-level module for LS-GAN in the output level.
2. We have not tried to use multi-level for LS-GAN. 
3. Yes, adding the multi-level module could improve the performance, but it will also need some hyperparameter tuning. If you intend to add it, I would suggest trying the default --lambda-seg 0.1 and --lambda-adv-target1 0.002 (proportional to the original setting) as the start to tune the hyperparameters.

**Comment by Nadavc220** [comment]
Thanks for the quick response.
